### PR TITLE
Allow users to change thresholds for evicting pods in VPA

### PR DESCRIPTION
**Pull request by maksim-paskal**
Allow users to change thresholds for evicting pods that use resources lower that recomended. Some pods needs to lower their request less than 12h.

This changes only allow users to change hardcoded values

New flags that will added with this PR
| Name | Default | Usage |
| --- | --- | --- |
| `pod-update-threshold` | 0.1 | Ignore updates that have priority lower than the value of this flag |
| `in-recommendation-bounds-eviction-lifetime-threshold` | 12h | Pods that live for at least that long can be evicted even if their request is within the [MinRecommended...MaxRecommended] range |

### CODE REVIEW

**Comment by maksim-paskal** [comment]
I signed it

File: vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go [hunk 1]
```diff
@@ -29,12 +29,13 @@ import (
 	"k8s.io/klog"
 )
 
-const (
+var (
 	// Ignore change priority that is smaller than 10%.
-	defaultUpdateThreshold = 0.10
+	defaultUpdateThreshold = flag.Float64("priority-update-threshold", 0.10, "Ignore change priority that is smaller this value")
```
Comment (jbartosik) [inline vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go:34]: Please make the flag description more descriptive.

Ignore updates that have priority lower than the value of this flag.

Short explanation of what is update priority would be useful too.

File: vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go [hunk 2]
```diff
@@ -29,12 +29,13 @@ import (
...
+
 	// Pods that live for at least that long can be evicted even if their
 	// request is within the [MinRecommended...MaxRecommended] range.
-	podLifetimeUpdateThreshold = time.Hour * 12
+	podLifetimeUpdateThreshold = flag.Duration("priority-lifetime-threshold", time.Hour*12, "Pods that live for at least that long can be evicted even if their request is within the [MinRecommended...MaxRecommended] range")
```
Comment (jbartosik) [inline vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go:38]: I think current name of this flag could be more descriptive. What do you think about `in-recommendation-bounds-eviction-lifetime-threshold`?

File: vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go [hunk 3]
```diff
@@ -29,12 +29,13 @@ import (
...
 )
 
 var (
```
Comment (jbartosik) [inline vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go:41]: Please merge those two `var` sections

File: vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go (hunk 2 above)
Comment (maksim-paskal) [reply vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go:38]: I think prefix `in-recommendation` is unnecessary

what do you think about this names with pod prefix?

| Name | Default | Usage |
| --- | --- | --- |
| `pod-update-threshold` | 0.1 | Ignore updates that have priority lower than the value of this flag |
| `pod-eviction-lifetime-threshold` | 12h | Pods that live for at least that long can be evicted even if their request is within the [MinRecommended...MaxRecommended] range |

Comment (jbartosik) [reply vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go]: I think that the `in-recommendatiotion-bounds` part is necessary (we will evict younger pods, if they are out of the [MinRecommended...MaxRecommended]  interval). 

`pod-eviction-lifetime-threshold` sounds like we wouldn't evict younger pods no matter what to me

Comment (maksim-paskal) [reply vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go]: ok - finally we got this flags. Ok?

| Name | Default | Usage |
| --- | --- | --- |
| `in-recommendation-update-threshold` | 0.1 | Ignore updates that have priority lower than the value of this flag |
| `in-recommendation-bounds-eviction-lifetime-threshold` | 12h | Pods that live for at least that long can be evicted even if their request is within the [MinRecommended...MaxRecommended] range |

Comment (jbartosik) [reply vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go]: Sounds good. Please update the PR

Comment (maksim-paskal) [reply vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go]: PR updated

Comment (jbartosik) [reply vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go]: Looks like I didn't explain myself clearly enough, sorry about that. I think that we need the `in-recommendation-bounds` prefix for the flag that controls after how long VPA will evict pods which have requests in the  `\[`MinRecommended`...`MaxRecommended`] interval.

We don't need the prefix for the flag which controls priority threshold, below which we will ignore updates.


So I'm suggesting

| Name | Default | Usage |
| --- | --- | --- |
| pod-update-threshold | 0.1 | Ignore updates that have priority lower than the value of this flag |
| in-recommendation-bounds-eviction-lifetime-threshold | 12h | Pods that live for at least that long can be evicted even if their request is within the [MinRecommended...MaxRecommended] range|

Comment (maksim-paskal) [reply vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator.go]: LGTM - PR updated

**Review by jbartosik — APPROVED** [review]
/lgtm
/approve

Thank you

**Pull request merged**